core/base/document.php

```diff
@@ -586,6 +587,27 @@ public function save( $data ) {
 		return true;
 	}
 
+	/**
+	 * @param $new_settings
+	 *
+	 * @return $this
+	 */
+	public function update_settings( $new_settings ) {
+		$document_settings = $this->get_meta( PageManager::META_KEY );
+
+		if ( ! $document_settings ) {
+			$document_settings = [];
+		}
+
+		SettingsManager::get_settings_managers( 'page' )
```

Comment:
What will happen if the new settings include external settings (such as post title, excerpt, status, etc.)?

Reply:
It will be included as current behavior.

Reply:
This function should call the `save_settings` method with the merged settings